src/connectors/kessel/impl.js

```diff
@@ -0,0 +1,227 @@
+            throw error;
+        }
+    }
+
+    extractWorkspaceId(identity) {
+        // Extract workspace/organization ID from identity
+        return identity.identity?.account_number ||
+               identity.identity?.org_id ||
+               identity.account_number ||
+               identity.org_id ||
+               'default';
+    }
+
+    getIdentityFromHeaders() {
+        const headers = this.getForwardedHeaders();
+
+        // Try to extract identity from x-rh-identity header
+        if (headers['x-rh-identity']) {
+            try {
+                const identityHeader = Buffer.from(headers['x-rh-identity'], 'base64').toString();
+                return JSON.parse(identityHeader);
+            } catch (error) {
+                log.warn({ error }, 'Failed to parse x-rh-identity header');
+            }
+        }
+
+        return null;
+    }
+
+    async ping () {
+        if (!this.kesselConfig.enabled) {
+            log.debug('Kessel not enabled, skipping ping');
+            return;
+        }
+
+        try {
+            await this.pingPermissionCheck();
+        } catch (error) {
+            log.warn({ error }, 'Kessel ping check failed');
+            assert(false, 'Kessel ping failed');
+        }
+    }
+
+    // Compatibility method to check specific permission
+    async hasPermission(resource, action, subject) {
+        if (!this.kesselConfig.enabled || !this.initialized || !this.kesselClient) {
+            return false;
+        }
+
+        const startTime = Date.now();
+        try {
+            // Convert traditional RBAC permission to workspace permission
+            const workspacePermission = this.convertRbacToWorkspacePermission(resource, action);
+            if (!workspacePermission) {
+                log.warn({ resource, action }, 'Unknown permission mapping');
+                return false;
+            }
+
+            // Get workspace ID from subject or use default
+            const workspaceId = this.getDefaultWorkspaceIdForSubject(subject);
+
+            // Check the specific permission using async/await
+            const allowed = await this.checkSinglePermission(subject, workspaceId, workspacePermission);
+
+            // Record successful metric
+            this.permissionMetrics.observe(Date.now() - startTime, 200);
+
+            return allowed;
+        } catch (error) {
+            // Record error metric
+            this.permissionMetrics.observe(Date.now() - startTime, 500);
+
+            log.error({ error, resource, action, subject }, 'Failed to check permission with Kessel');
+            return false;
+        }
+    }
+
+    convertRbacToWorkspacePermission(resource, action) {
+        // Convert traditional RBAC permission to workspace permission
+        // Pattern: ${resource}:${action} -> remediations_${action}_${resource}
+        return `remediations_${action}_${resource}`;
```

Comment:
This will need some updates as well - we use different verbs e.g. `read` vs `view`